pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -309,6 +307,36 @@ private void messageReceived(ConsumerImpl<T> consumer, Message<T> message) {
         }
     }
 
+    void notifyPendingBatchReceivedCallBack() {
+        OpBatchReceive<T> opBatchReceive = pendingBatchReceives.poll();
+        if (opBatchReceive == null || opBatchReceive.future == null) {
+            return;
+        }
+        notifyPendingBatchReceivedCallBack(opBatchReceive);
+    }
+
+    void notifyPendingBatchReceivedCallBack(OpBatchReceive<T> opBatchReceive) {
+        MessagesImpl<T> messages = new MessagesImpl<>(batchReceivePolicy.getMaxNumMessages(),
+                batchReceivePolicy.getMaxNumBytes());
+        Message<T> msgPeeked = incomingMessages.peek();
```

Comment:
I see the similar pattern in a few places. Can we make a function for it?

Reply:
I have added a method getReUseableMessagesImpl() to avoid create lots of MessagesImpl.

Reply:
fixed